src/main/groovy/com/bmuschko/gradle/vagrant/tasks/Vagrant.groovy

```diff
@@ -23,34 +23,35 @@ import com.bmuschko.gradle.vagrant.utils.OsUtils
 import groovy.transform.PackageScope
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.InputDirectory
-import org.gradle.api.tasks.Internal
-import org.gradle.api.tasks.PathSensitive
-import org.gradle.api.tasks.PathSensitivity
-import org.gradle.api.tasks.TaskAction
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.tasks.*
 
-class Vagrant extends DefaultTask {
+abstract class Vagrant extends DefaultTask {
     static final String TASK_GROUP = 'Vagrant'
 
     /**
      * The Vagrant command to run.
      */
     @Input
-    List<String> commands
+    abstract ListProperty<String> getCommands()
+
+    @Internal
+    abstract ListProperty<String> getOptions()
```

Comment:
I think it's easier to understand to have this also marked as an input although it's implicitly tracked by the inputs in the subclasses that use it.